lib/block_reservation.rb

```diff
@@ -0,0 +1,21 @@
+module BookingSystem
+  class BlockReservation < BookingSystem::Reservation
+
+    attr_reader :room, :reservation_dates, :price
+
+    DISCOUNT_RATE = 0.15
+
+    def initialize(room, check_in, check_out, price)
+      super
```

Comment:
Does this need to be its own class? Seems like you could modify `Reservation` and to accept an optional `discount_rate` in the constructor with a default value of 0, then use it both places.